tests/rptest/scale_tests/many_partitions_test.py

```diff
@@ -989,3 +997,530 @@ def progress_check():
                     f"Expected throughput {expect_mbps:.2f}MiB/s, got throughput {actual_mbps:.2f}MiB/s"
                 )
                 raise
+
+    def _try_parse_json(self, node, jsondata):
+        try:
+            return json.loads(jsondata)
+        except ValueError:
+            self.logger.debug(
+                f"{str(node.account)}: Could not parse as json: {str(jsondata)}"
+            )
+            return None
+
+    def _write_and_random_read_many_topics(self, num_topics, topic_names):
+        """
+            Test checks that each of the many topics can be written to.
+            This produce/consume implementation will check actual data
+            of the messages to ensure that all of the messages are delivered
+
+            Pick X number of topics, write 1000 messages in each
+            Pick random one among them, consume all messages
+            Iterate.
+        """
+        def _send_messages(topic):
+            """
+                Simple function that sends indices as messages
+            """
+            errors = []
+
+            def acked(err, msg):
+                """
+                    Simple and unsafe callback
+                """
+                if err is not None:
+                    errors.append(f"FAIL: {str(msg)}: {str(err)}")
+
+            # Sent indices as values
+            p = kclient.get_producer()
+            sent_count = 0
+            for idx in range(1, message_count + 1):
+                # Async message sending func
+                p.produce(topic,
+                          key=f"key_{idx}",
+                          value=f"{idx:04}",
+                          callback=acked)
+                # Make sure message sent, aka sync
+                p.flush()
+                sent_count += 1
+            # Return stats and errors
+            return sent_count, errors
+
+        def check_consecutive(numbers_list):
+            n = len(numbers_list) - 1
+            # Calculate iterative difference for the array. It should be 1.
+            return (sum(numpy.diff(sorted(numbers_list)) == 1) >= n)
+
+        # Prepare librdkafka python client
+        kclient = PythonLibrdkafka(self.redpanda)
```

Comment:
Should probably look this up myself, but does `kclient` create a new producer/consumer every time you call `kclient.get_producer()` and `kclient.get_consumer()`? Also are these methods thread-safe?

Reply:
From what I found, they are thread safe and actually a python interface to librdkafka (confluent_kafka module for python)